src/Compilers/VisualBasic/Portable/Lowering/LocalRewriter/LocalRewriter_ConditionalExpressions.vb

```diff
@@ -218,6 +218,14 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                 Return rewrittenRight
             End If
 
+            ' Optimize If(left, right) to left.GetValueOrDefault() when left is T? and right is the default value of T
+            If rewrittenLeft.Type.IsNullableType AndAlso 
+               rewrittenRight.IsDefaultValue AndAlso
+               rewrittenRight.Type.IsSameTypeIgnoringAll(rewrittenLeft.Type.GetNullableUnderlyingType) _
+            Then
+                Return NullableValueOrDefault(rewrittenLeft)
```

Comment:
>Return NullableValueOrDefault(rewrittenLeft) [](start = 16, length = 44)

Is this going to interfere with conditional access optimization below? We probably should let that the priority. Please add relevant unit test(s) and for similar scenarios in C#.

Reply:
It looks like it does interfere and the failing test Microsoft.CodeAnalysis.VisualBasic.UnitTests.Semantics.ConditionalAccessTests.CodPlexBug470 shows that. We probably want to preserve original behavior for that scenario

---
In reply to: [197275118](https://github.com/dotnet/roslyn/pull/27700#discussion_r197275118) [](ancestors = 197275118)

Reply:
You're right, it was indeed interfering, I put the optimization too early.
I've added tests for both C# and VB and fixed the broken VB case.